### DISCUSSION TITLE
UI design

### DISCUSSION THREAD

**Discussion by the-rivulet** [oskarrough/slaytheweb#155] [General]
Hi, I was creating a similar game to Slay the Web called CoSTS (Clone of Slay the Spire) when I noticed this. The UI looks great, can you help me with making something similar? My code is [here](http:/github.com/wooga123/wooga123.github.io/blob/main/costs.html) and it is hosted [here](http:/wooga123.github.io/costs.html).

**Comment by oskarrough** [comment]
That's so cool haha! Love to see how you did it. Well done.

I would like to help if I can. The UI for the "my version" here is made with two dependencies: preact and [htm](https://github.com/developit/htm). I basically have a huge `Object` called `state` where everything is stored. The UI renders the state. Whenever you do something in the game, it's an "action" and will update the state, render again. For me this pattern works well, maybe it will for you as well.

You can read more about it here https://github.com/oskarrough/slaytheweb/blob/main/DOCUMENTATION.md

If you're interested I'd of course be very excited to see more collaborators on Slay the Web as well..
